iree/hal/dylib/dylib_executable.cc

```diff
@@ -34,9 +34,12 @@ DyLibExecutable::DyLibExecutable() = default;
 
 DyLibExecutable::~DyLibExecutable() {
   IREE_TRACE_SCOPE0("DyLibExecutable::dtor");
-  // TODO(benvanik): move to an atexit handler when tracing is enabled.
-  // executable_library_.release();
+#if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION
+  // Leak the library when tracing, since the profiler may still be reading it.
+  executable_library_.release();
```

Comment:
The reason I had still TODO'd this was because I want to make sure msan is still ok with this happening. An atexit handler that actually released things, the msan annotation to disable leak checker on a pointer, etc are what we need to play with. This is fine for the branch but you should leave the TODO as it's not really a solution that will work.

Also, we need the CI to run msan (/asan/etc) as well as with tracing flags enabled - none of this is covered right now and that's silly.

Reply:
Trying to run under MSAN now...

* We don't have an MSAN Bazel config (only [ASAN](https://github.com/google/iree/blob/5a22d1bf3b77c83ec9ab8b1e183b50225e03b7a2/build_tools/bazel/iree.bazelrc#L152-L176))
* Using our [CMake config](https://github.com/google/iree/blob/5a22d1bf3b77c83ec9ab8b1e183b50225e03b7a2/CMakeLists.txt#L230), I get blocking errors just trying to build `iree_tools_iree-translate` or `iree_tools_iree-run-module`. First [here](https://github.com/google/iree/blob/5a22d1bf3b77c83ec9ab8b1e183b50225e03b7a2/build_tools/embed_data/generate_cc_embed_data_main.cc#L118), then in `mlir-tblgen` and `llvm-tblegen` somewhere

I'll spend a bit more time on it and see if I can fix the current issues. [atexit](https://man7.org/linux/man-pages/man3/atexit.3.html) seems reasonable, or the TODO could be added back.

Reply:
Well, running MSAN and ASAN was tricky, but I don't see any warnings/errors for `executable_library_.release();`.

There are a few things in https://github.com/abseil/abseil-cpp/blob/master/absl/debugging/leak_check.h we could use if it later becomes an issue.

I'm not concerned about this *if* we only trace in development builds. If we determine that tracing is cheap enough to always enable then we should care much more :P

WDYT?

Reply:
Restored the TODO with a bit more info.